### DISCUSSION TITLE
Converting point cloud to .stl file

### DISCUSSION THREAD

**Discussion by Ali1990dashti** [pyvista/pyvista#4961] [Q&A] [answered]
Dear community,
I just started using PyVista and have a very simple question. How can I convert a point cloud into a `.stl` file?
I am using PyVista `0.30.1`. The following code gives me the `.vtk` file but I need the `.stl` file:
```python
import pyvista as pv
from numpy import genfromtxt
F2 = genfromtxt('surface.csv', delimiter=',')
point_cloud = pv.PolyData(F2)
surf = point_cloud.delaunay_3d()
surf.save('mesh.vtk')
```
The point cloud is also attached as a `.csv` file.

Thanks for any help.
Cheers

[surface.csv](https://github.com/pyvista/pyvista/files/12736108/surface.csv)

**Comment by tkoyama010** [comment, marked as answer]
[pv.save_meshio](https://docs.pyvista.org/version/stable/api/utilities/_autosummary/pyvista.save_meshio.html#pyvista-save-meshio) is what you are looking for.
```python
>>> pv.save_meshio("mesh.stl", surf)
Warning: STL can only write triangle cells. No triangle cells found.
```
However, the mesh you have created cannot be saved in stl because it consists of tetrahedra. You need to create a triangle mesh.
```python
>>> np.all(surf.celltypes == pv.CellType.TETRA)
True
```

**Reply by Ali1990dashti** [reply]
Dear @tkoyama010 

Thanks for your hint. I changed the mesh from 3d to 2d and it worked:

```python
import pyvista as pv
from numpy import genfromtxt
F2 = genfromtxt('surface.csv', delimiter=',')
point_cloud = pv.PolyData(F2)
surf = point_cloud.delaunay_2d()
pv.save_meshio("mesh.stl", surf)
```

Thanks a lot for your support.

**Reply by MatthewFlamm** [reply]
Also the normal `PolyData.save` method is documented as supporting stl as a format. https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyData.save.html#pyvista-polydata-save

**Reply by Ali1990dashti** [reply]
Dear @MatthewFlamm 
Thanks for your comment. I already saw this page and learned how PyVista exports the `Sphere `as a `stl `file, but I could not figure it how to export my own surface as  `stl ` file. In fact, I only have the point cloud and want to make an `stl `out of it.
